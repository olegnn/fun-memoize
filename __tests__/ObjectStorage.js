const { ObjectStorage } = require("../build/storage/ObjectStorage");
const { isPrimitiveValue } = require("../build/value");
const { TRICKY_VALUES, createBasicStorageTests } = require("./helpers");

const keys = [
  1,
  "234",
  13123123,
  12.23123,
  12.22,
  123131231,
  123131231n,
  12313123123123123n,
  true,
  ...TRICKY_VALUES,
].filter(isPrimitiveValue);

describe("ObjectStorage", () => {
  createBasicStorageTests(ObjectStorage, keys);

  it("checks non-primitive errors", () => {
    const storage = new ObjectStorage();

    for (const key of [
      {},
      () => {},
      function () {},
      function* () {},
      Object.create(null),
    ]) {
      expect(() => storage.set(key, 1)).toThrowErrorMatchingSnapshot();
      expect(() => storage.get(key)).toThrowErrorMatchingSnapshot();
      expect(() => storage.drop(key)).toThrowErrorMatchingSnapshot();
      expect(() => storage.has(key)).toThrowErrorMatchingSnapshot();
    }
  });
});
