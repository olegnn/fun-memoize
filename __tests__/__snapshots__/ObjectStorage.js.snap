// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectStorage checks non-primitive errors 1`] = `"Invalid value [object Object] with type object, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 2`] = `"Invalid value [object Object] with type object, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 3`] = `"Invalid value [object Object] with type object, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 4`] = `"Invalid value [object Object] with type object, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 5`] = `"Invalid value () => {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 6`] = `"Invalid value () => {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 7`] = `"Invalid value () => {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 8`] = `"Invalid value () => {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 9`] = `"Invalid value function () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 10`] = `"Invalid value function () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 11`] = `"Invalid value function () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 12`] = `"Invalid value function () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 13`] = `"Invalid value function* () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 14`] = `"Invalid value function* () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 15`] = `"Invalid value function* () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 16`] = `"Invalid value function* () {} with type function, expected primitive"`;

exports[`ObjectStorage checks non-primitive errors 17`] = `"Cannot convert object to primitive value"`;

exports[`ObjectStorage checks non-primitive errors 18`] = `"Cannot convert object to primitive value"`;

exports[`ObjectStorage checks non-primitive errors 19`] = `"Cannot convert object to primitive value"`;

exports[`ObjectStorage checks non-primitive errors 20`] = `"Cannot convert object to primitive value"`;
